evm/src/main/java/org/hyperledger/besu/evm/precompile/PrecompiledContract.java

```diff
@@ -52,5 +54,90 @@
    * @param messageFrame context for this message
    * @return the output of the pre-compiled contract.
    */
-  Bytes compute(Bytes input, @Nonnull MessageFrame messageFrame);
+  @SuppressWarnings("deprecation")
+  default PrecompileContractResult computePrecompile(
+      final Bytes input, @Nonnull final MessageFrame messageFrame) {
+    final Bytes result = compute(input, messageFrame);
+    if (result == null) {
+      return PrecompileContractResult.halt(null, Optional.of(ExceptionalHaltReason.NONE));
+    } else {
+      return PrecompileContractResult.success(result);
+    }
+  }
+
+  /**
+   * Executes the pre-compiled contract.
+   *
+   * @param input the input for the pre-compiled contract.
+   * @param messageFrame context for this message
+   * @return the output of the pre-compiled contract.
+   * @deprecated Migrate to use {@link #computePrecompile(Bytes, MessageFrame)}.
+   */
+  @Deprecated(since = "22.1.2")
+  default Bytes compute(final Bytes input, final @Nonnull MessageFrame messageFrame) {
```

Comment:
If an implementing class fails to override compute, won't we get a call loop between these default implementations?

Reply:
This is the side effect of backwards compatability.

First, to support pre-existing precompiles we cannot require the new `computePrecompile` method be implemented, so it must have a default method.

Second, we don't want to require new precompiles to stub out a deprecated method useless code.  So for good ergonomics we need a default.

Reply:
both of these defaults call the other, but if we only provided a "real" implementation for one and an error producing one for the other then we still only discover at runtime that you didn't override methods.  Even if we detect the loop (which has performance impacts for well behaving subclasses) you still won't detect the failure except at runtime.  And by "at runtime" I mean in unit test code because everyone unit tests before they do a test network right?

Reply:
So the question is _how_ to correct it.  Both methods must provide defaults.  If we make computePrecomile the error throwing method then that breaks backwards compatibility.  If we make compute the error throwing method it breaks downstream users that still call compute directly.  If we presume that implementors are useful and implement at least one of the methods there is no loop.

So as the reviewer, how do you want me to break things?

Reply:
> If we presume that implementors are useful and implement at least one of the methods there is no loop.

Cool, thats all i needed to know.